Modules/_datetimemodule.c

```diff
@@ -6540,23 +6521,65 @@ PyInit__datetime(void)
     Py_BUILD_ASSERT(DI100Y == 25 * DI4Y - 1);
     assert(DI100Y == days_before_year(100+1));
 
-    us_per_ms = PyLong_FromLong(1000);
-    us_per_second = PyLong_FromLong(1000000);
-    us_per_minute = PyLong_FromLong(60000000);
-    seconds_per_day = PyLong_FromLong(24 * 3600);
+    if (!us_per_ms) {
+        us_per_ms = PyLong_FromLong(1000);
+    }
+    if (!us_per_second) {
+        us_per_second = PyLong_FromLong(1000000);
+    }
+    if (!us_per_minute) {
+        us_per_minute = PyLong_FromLong(60000000);
+    }
+    if (!seconds_per_day) {
+        seconds_per_day = PyLong_FromLong(24 * 3600);
+    }
     if (us_per_ms == NULL || us_per_second == NULL ||
         us_per_minute == NULL || seconds_per_day == NULL)
-        return NULL;
+        return -1;
 
     /* The rest are too big for 32-bit ints, but even
      * us_per_week fits in 40 bits, so doubles should be exact.
      */
-    us_per_hour = PyLong_FromDouble(3600000000.0);
-    us_per_day = PyLong_FromDouble(86400000000.0);
-    us_per_week = PyLong_FromDouble(604800000000.0);
+    if (!us_per_hour) {
+        us_per_hour = PyLong_FromDouble(3600000000.0);
+    }
+    if (!us_per_day) {
+        us_per_day = PyLong_FromDouble(86400000000.0);
+    }
+    if (!us_per_week) {
+        us_per_week = PyLong_FromDouble(604800000000.0);
+    }
     if (us_per_hour == NULL || us_per_day == NULL || us_per_week == NULL)
-        return NULL;
-    return m;
+        return -1;
+
+    // Workaround to allow static PyTypeObjects to have they dict redefined.
+    // XXX: to be removed once _datetime moves to heap allocated PyTypeObjects.
```

Comment:
Are there negative consequences to this if we *don't* move to heap allocated PyTypeObjects?

Are there any downsides to using heap allocated types?

I don't want us to be in a situation where we have some workaround with negative consequences that we end up not being able to remove because we can't convert to heap allocated types.